ably/options.go

```diff
@@ -14,36 +15,72 @@ const (
 	ProtocolMsgPack = "msgpack"
 )
 
+var errInvalidKey = errors.New("invalid key format")
+
+var DefaultOptions = &ClientOptions{
+	RestEndpoint:     "https://rest.ably.io",
+	RealtimeEndpoint: "wss://realtime.ably.io:443",
+	Protocol:         ProtocolJSON, // TODO: make it ProtocolMsgPack
+}
+
 type ClientOptions struct {
-	RealtimeEndpoint string
 	RestEndpoint     string
-	Key              string
+	RealtimeEndpoint string
 	Token            string
 	Secret           string
 	ClientID         string
-	Protocol         string
+	Protocol         string // either ProtocolJSON or ProtocolMsgPack
 	UseTokenAuth     bool
-	NoTLS            bool
 
 	HTTPClient *http.Client
 }
 
-func (p *ClientOptions) Prepare() {
-	if p.Key != "" {
-		p.parseKey()
+var (
+	restTLS       = strings.NewReplacer("http://", "https://")
+	restNoTLS     = strings.NewReplacer("https://", "http://")
+	realtimeTLS   = strings.NewReplacer("ws://", "wss://", ":443", ":80")
+	realtimeNoTLS = strings.NewReplacer("wss://", "ws://", ":80", ":443")
+)
+
+func (opts *ClientOptions) SetTLS(enabled bool) {
+	if enabled {
+		opts.RestEndpoint = restTLS.Replace(opts.RestEndpoint)
+		opts.RealtimeEndpoint = realtimeTLS.Replace(opts.RealtimeEndpoint)
+	} else {
+		opts.RestEndpoint = restNoTLS.Replace(opts.RestEndpoint)
+		opts.RealtimeEndpoint = realtimeNoTLS.Replace(opts.RealtimeEndpoint)
 	}
```

Comment:
It's just a fix, the REST client was implemented that way the Endpoints are URI instead of hostnames, but API did not provide method to switch between TLS/no TLS; I'll rewrite it to make Endpoints just hostnames.